app/src/main/java/com/example/nigeriatelemedicineapp/utils/DateUtlis.kt

```diff
@@ -0,0 +1,59 @@
+package com.example.nigeriatelemedicineapp.utils
+
+class DateUtlis {
+
+    companion object {
+
+        fun checkDateFormat(date: String) : Boolean {
+            return DateUtlis().validate(date)
+        }
+
+        fun getCorrectDateFormat(date:String) : String{
+           val StringArray = date.split("-")
+            var correctFormat: String=""
+            correctFormat= correctFormat+StringArray[2]+"-"+StringArray[1]+"-"+StringArray[0]
+            return correctFormat.trim()
+        }
+
+    }
+
+    fun validate(dateString: String): Boolean {
```

Comment:
Is there a java library that can handle this for us? Seems like a lot of opportunity for error.

Reply:
Actually, the in-built classes limit our checks. Many things like if month is greater than 12 or not are not checked by it. Plus if rigorously tested the function.  The best I can do here is put it in a try catch block . That eliminates any possibility of unchecked cases we might have forgotten.

Also this idea of writing the whole code by myself was taken form OpenMRS android client which does something like this only.

Reply:
Which in-built classes are you using? Can we use the java.text package (https://docs.oracle.com/javase/8/docs/api/java/text/package-summary.html)?

Example here: https://kodejava.org/how-do-i-check-if-a-string-is-a-valid-date/

Let's discuss at our next meeting if you have more questions.